### PR TITLE
Dasd 8817 migrate generate config task

### PR DESCRIPTION
**Pull request by NickGraham101**

### CODE REVIEW

File: azure-pipelines-templates/deploy/step/generate-config.yml [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
+  TargetFileName: '*.schema.json'
+  ConfigurationSecrets: {}
+  TableName: 'Configuration'
 
 steps:
-- task: apprenticeship-service.GenerateEnvironmentConfiguration.custom-build-task.GenerateEnvironmentConfiguration@3
-  displayName: 'Generate Configuration'
+- checkout: das-platform-automation
+- task: AzurePowerShell@5
+  displayName: New-ConfigurationTableEntry
```
Comment (nbowes24) [inline azure-pipelines-templates/deploy/step/generate-config.yml:11]: What do you think about this being 'Generate Configuration' to match what we're used to seeing on the release steps

File: azure-pipelines-templates/deploy/step/generate-config.yml [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
-    StorageAccountName: '$(ConfigurationStorageAccountName)'
-    TableName: ${{ parameters.TableName }}
+    azureSubscription: ${{ parameters.ServiceConnection }}
+    scriptPath: das-platform-automation/Infrastructure-Scripts/New-ConfigurationTableEntry/New-ConfigurationTableEntry.ps1
+    scriptArguments:
+      -SourcePath ${{ parameters.SourcePath }}
+      -TargetFilename ${{ parameters.TargetFileName }}
+      -StorageAccountName '$(ConfigurationStorageAccountName)'
+      -StorageAccountResourceGroup '$(SharedEnvResourceGroup)'
+      -EnvironmentName '$(EnvironmentName)'
```
Comment (nbowes24) [inline azure-pipelines-templates/deploy/step/generate-config.yml:20]: Should these 3 be parameters?

File: azure-pipelines-templates/deploy/step/generate-config.yml (hunk 1 above)
Comment (NickGraham101) [reply azure-pipelines-templates/deploy/step/generate-config.yml:11]: Improved displayName

File: azure-pipelines-templates/deploy/step/generate-config.yml (hunk 2 above)
Comment (jack-cloud-platform) [reply azure-pipelines-templates/deploy/step/generate-config.yml:20]: I think they should ideally but the migration efforts to pass in the additional parameters for the use of this template would not be worth it.

Maybe can be resolved when moving away from this configuration method.

**Review by nbowes24 — APPROVED** [review]

File: azure-pipelines-templates/deploy/step/generate-config.yml (hunk 2 above)
Comment (NickGraham101) [reply azure-pipelines-templates/deploy/step/generate-config.yml]: Changed to params

**Review by nbowes24 — APPROVED** [review]

**Pull request merged**